index.js

```diff
@@ -196,14 +196,28 @@
     return _funcPath(false, path, implementations);
   }
 
+  //  functionName :: Function -> String
+  var functionName = 'name' in function f() {} ?
+    function functionName(f) { return f.name; } :
+    /* istanbul ignore next */
+    function functionName(f) {
+      var match = /function (\w*)/.exec(f);
+      return (match || '') && match[1];
```

Comment:
I think `match == null ? '' : match[1]` would be clearer. Currently both `||` and `&&` are "applied" in the failure case.